From 53bc04c1813581546077d563147045b9856a65d5 Mon Sep 17 00:00:00 2001
From: Mark Wharton <marwha@amazon.com>
Date: Tue, 2 Jun 2020 23:04:54 -0700
Subject: [PATCH 1/2] Smart Screen SDK for Alexa Auto SDK

---
 modules/Alexa/CMakeLists.txt | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/modules/Alexa/CMakeLists.txt b/modules/Alexa/CMakeLists.txt
index cb8629a..eb2c52f 100644
--- a/modules/Alexa/CMakeLists.txt
+++ b/modules/Alexa/CMakeLists.txt
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
 project(AlexaModule LANGUAGES CXX)
 
 find_package(PkgConfig)
-pkg_check_modules(ASDK REQUIRED AlexaClientSDK=1.19.0)
+pkg_check_modules(ASDK REQUIRED AlexaClientSDK=1.19.1)
 
 include(build/BuildDefaults.cmake)
 include(tools/Testing.cmake)
@@ -12,13 +12,13 @@ include(tools/Testing.cmake)
 include(build/cmake/PrepareInstall.cmake)
 
 # Alexa Client SDK targets.
-add_subdirectory("ThirdParty")
+# add_subdirectory("ThirdParty")
 add_subdirectory("SmartScreenCapabilityAgents")
 add_subdirectory("SmartScreenSDKInterfaces")
-add_subdirectory("ApplicationUtilities")
-add_subdirectory("SampleApp")
-add_subdirectory("doc")
-add_subdirectory("Utils")
+# add_subdirectory("ApplicationUtilities")
+# add_subdirectory("SampleApp")
+# add_subdirectory("doc")
+# add_subdirectory("Utils")
 
 # Create .pc pkg-config file
 include(build/cmake/GeneratePkgConfig.cmake)
-- 
2.21.1 (Apple Git-122.3)

