From 250a97ca9f327e5ea6b508193e0ccbf15017abd1 Mon Sep 17 00:00:00 2001
From: Patrick Tsai <yingtsai@amazon.com>
Date: Mon, 13 Sep 2021 12:53:39 -0700
Subject: [PATCH 1/1] Fix linker error of _getpid for QNX

---
 lib/smb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/smb.c b/lib/smb.c
index 12f9925..38cf06e 100644
--- a/lib/smb.c
+++ b/lib/smb.c
@@ -34,7 +34,7 @@
 #include <process.h>
 #ifdef CURL_WINDOWS_APP
 #define getpid GetCurrentProcessId
-#elif !defined(MSDOS)
+#elif !defined(MSDOS) && !defined(__QNXNTO__)
 #define getpid _getpid
 #endif
 #endif
--
2.33.0

From 6f6f802aef31414d2cc572f7f867b6eb7b1a47fb Mon Sep 17 00:00:00 2001
From: Patrick Tsai <yingtsai@amazon.com>
Date: Mon, 21 Feb 2022 16:20:06 -0800
Subject: [PATCH] Fix linker error of _getpid for QNX

---
 lib/smb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/smb.c b/lib/smb.c
index fd49cf6..4ccc764 100644
--- a/lib/smb.c
+++ b/lib/smb.c
@@ -32,7 +32,7 @@
 #include <process.h>
 #ifdef CURL_WINDOWS_APP
 #define getpid GetCurrentProcessId
-#elif !defined(MSDOS)
+#elif !defined(MSDOS && !defined(__QNXNTO__))
 #define getpid _getpid
 #endif
 #endif
--
2.35.1