From 8e43a51edd46f96b26b0b4941a24a936c169e141 Mon Sep 17 00:00:00 2001
From: Eze Cervantes <ezeqc@amazon.com>
Date: Tue, 23 Nov 2021 09:42:06 -0800
Subject: [PATCH 05/14] Template runtime version 1.2

---
 .../TemplateRuntime/src/TemplateRuntime.cpp                     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/Alexa/SmartScreenCapabilityAgents/TemplateRuntime/src/TemplateRuntime.cpp b/modules/Alexa/SmartScreenCapabilityAgents/TemplateRuntime/src/TemplateRuntime.cpp
index ab74f98..fb8d2ec 100644
--- a/modules/Alexa/SmartScreenCapabilityAgents/TemplateRuntime/src/TemplateRuntime.cpp
+++ b/modules/Alexa/SmartScreenCapabilityAgents/TemplateRuntime/src/TemplateRuntime.cpp
@@ -42,7 +42,7 @@ static const std::string TEMPLATERUNTIME_CAPABILITY_INTERFACE_TYPE = "AlexaInter
 /// TemplateRuntime interface name
 static const std::string TEMPLATERUNTIME_CAPABILITY_INTERFACE_NAME = "TemplateRuntime";
 /// TemplateRuntime interface version
-static const std::string TEMPLATERUNTIME_CAPABILITY_INTERFACE_VERSION = "1.1";
+static const std::string TEMPLATERUNTIME_CAPABILITY_INTERFACE_VERSION = "1.2";
 
 /// String to identify log entries originating from this file.
 static const std::string TAG{"TemplateRuntime"};
-- 
2.36.0

