From 4ebd236fa6490a6c07541638d763b630251b951b Mon Sep 17 00:00:00 2001
From: Eze Cervantes <ezeqc@amazon.com>
Date: Fri, 3 Dec 2021 14:52:44 -0800
Subject: [PATCH] Ignore template runtime token

---
 .../TemplateRuntime/src/TemplateRuntime.cpp                 | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/modules/Alexa/SmartScreenCapabilityAgents/TemplateRuntime/src/TemplateRuntime.cpp b/modules/Alexa/SmartScreenCapabilityAgents/TemplateRuntime/src/TemplateRuntime.cpp
index 08e6fc1..fc1af08 100644
--- a/modules/Alexa/SmartScreenCapabilityAgents/TemplateRuntime/src/TemplateRuntime.cpp
+++ b/modules/Alexa/SmartScreenCapabilityAgents/TemplateRuntime/src/TemplateRuntime.cpp
@@ -661,13 +661,11 @@ void TemplateRuntime::executeRenderPlayerInfoCallbacks(bool isClearCard) {
 }
 
 void TemplateRuntime::executeRenderTemplateCallbacks(bool isClearCard) {
-    ACSDK_DEBUG3(LX("executeRenderTemplateCallbacks").d("isClear", isClearCard ? "True" : "False"));
+    ACSDK_DEBUG3(LX("executeRenderTemplateCallbacks").d("isClear", isClearCard ? "True" : "False").d("token", m_nonPlayerInfoCardToken));
     for (auto& observer : m_observers) {
         if (isClearCard) {
             executeNonPlayerInfoCardCleared(NonPlayerInfoDisplayType::RENDER_TEMPLATE);
-            if (!m_nonPlayerInfoCardToken.empty()) {
-                observer->clearTemplateCard(m_nonPlayerInfoCardToken);
-            }
+            observer->clearTemplateCard(m_nonPlayerInfoCardToken);
 
             m_nonPlayerInfoCardToken = "";
         } else {
-- 
2.30.1 (Apple Git-130)

